dd-trace-api/src/main/java/datadog/trace/api/SQLCommenter.java

```diff
@@ -0,0 +1,317 @@
+   * @return true if no custom fields are set
+   */
+  public static boolean isEmpty() {
+    return getContext().isEmpty();
+  }
+
+  /**
+   * Gets a copy of the current thread's SQL comment context map.
+   *
+   * @return a copy of the current context map, or null if empty
+   */
+  public static Map<String, String> getCopyOfContextMap() {
+    return getContext().getCopyOfContextMap();
+  }
+
+  /**
+   * Executes a block of code with the given custom fields, restoring the original context
+   * afterward.
+   *
+   * @param customFields the custom fields to use during execution
+   * @param callable the code block to execute
+   * @param <T> the return type of the callable
+   * @return the result of executing the callable
+   * @throws Exception if the callable throws an exception
+   */
+  public static <T> T withCustomFields(Map<String, Object> customFields, Callable<T> callable)
+      throws Exception {
+    return getContext().withSQLCommentFields(customFields, callable::call);
+  }
+
+  /**
+   * Executes a block of code with the given custom fields, restoring the original context
+   * afterward.
+   *
+   * @param callable the code block to execute
+   * @param keyValuePairs alternating key-value pairs to add to the context
+   * @param <T> the return type of the callable
+   * @return the result of executing the callable
+   * @throws Exception if the callable throws an exception
+   */
+  public static <T> T withCustomFields(Callable<T> callable, Object... keyValuePairs)
+      throws Exception {
+    return getContext().withSQLCommentFields(callable::call, keyValuePairs);
+  }
+
+  /**
+   * Executes a runnable with the given custom fields, restoring the original context afterward.
+   *
+   * @param customFields the custom fields to use during execution
+   * @param runnable the code block to execute
+   */
+  public static void withCustomFields(Map<String, Object> customFields, Runnable runnable) {
+    try {
+      withCustomFields(
+          customFields,
+          () -> {
+            runnable.run();
+            return null;
+          });
+    } catch (Exception e) {
+      if (e instanceof RuntimeException) {
+        throw (RuntimeException) e;
+      }
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Executes a runnable with the given custom fields, restoring the original context afterward.
+   *
+   * @param runnable the code block to execute
+   * @param keyValuePairs alternating key-value pairs to add to the context
+   */
+  public static void withCustomFields(Runnable runnable, Object... keyValuePairs) {
+    try {
+      withCustomFields(
+          () -> {
+            runnable.run();
+            return null;
+          },
+          keyValuePairs);
+    } catch (Exception e) {
+      if (e instanceof RuntimeException) {
+        throw (RuntimeException) e;
+      }
+      throw new RuntimeException(e);
+    }
+  }
+
+  // Direct access to the context - simpler than the provider pattern
```

Comment:
Hopefully this ends up being the same class from the same classloader? I guess if it works it works.